From 3dfbe36c5da03ee3c58bd250b754c1667ac72cae Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Wed, 20 Jan 2016 16:33:54 +0800
Subject: [PATCH 1600/5242] MLK-12294 tty: serial: imx: init sport->flags
 before tx DMA ready

commit  f007caac6483357f7202c61da59cde47e3458cdd from
https://source.codeaurora.org/external/imx/linux-imx.git

Before then DMA tx path, init the DMA tx path synchronal flags
sport->flags.

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/tty/serial/imx.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index fe66880..1bde76b 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1319,9 +1319,8 @@ static void imx_uart_enable_dma(struct imx_port *sport)
 {
 	u32 ucr1, ucr4;
 
-	imx_uart_setup_ufcr(sport, TXTL_DMA, RXTL_DMA);
-
 	init_waitqueue_head(&sport->dma_wait);
+	sport->flags = 0;
 
 	/* set UCR1 */
 	ucr1 = imx_uart_readl(sport, UCR1);
-- 
1.7.9.5

